src/ua/griddynamics/Purchase.java

```diff
@@ -0,0 +1,179 @@
+                    } else {
+                        for (Product p : c.getPRODUCTS()) {
+                            totalSum += p.getPRICE();
+                            System.out.println(p);
+                        }
+                        System.out.println("Total sum: $" + String.format("%.2f", totalSum));
+                    }
+                }
+            }
+        }
+    }
+
+   public static void showAllSortedPurchases() {
+        if (!noPurchasesAtAll()) {
+            List<Product> productsList = new ArrayList<>();
+            double totalSum = 0.0;
+            for (Category c : ALL_PRODUCTS) {
+                totalSum += c.getTotalSum();
+                productsList.addAll(c.getPRODUCTS());
+            }
+            System.out.println("\nAll: ");
+            productsList.stream()
+                    .sorted(Comparator.comparing(Product::getPRICE).reversed())
+                    .forEach(System.out::println);
+            System.out.println("Total sum: $" + String.format("%.2f", totalSum));
+        }
+    }
+
+    public static void showSortByCategories() {
+        System.out.println("\nTypes: ");
+        double totalSumOfAll = 0.00;
+        Map<String, Double> mapForSorting = new HashMap<>();
+        for (Category c : ALL_PRODUCTS) {
+            totalSumOfAll += c.getTotalSum();
+            if (c.getPRODUCTS().isEmpty()) {
+                mapForSorting.put(c.getType(), 0.00);
+            } else {
+                double totalSum = 0;
+                for (Product p : c.getPRODUCTS()) {
+                    totalSum += p.getPRICE();
+                }
+                mapForSorting.put(c.getType(), totalSum);
+            }
+        }
+        mapForSorting.entrySet().stream().sorted(Map.Entry.comparingByValue(Comparator.reverseOrder()))
+                .forEachOrdered(x -> System.out.println(x.getKey() + " $" + x.getValue()));
+        System.out.println("Total sum: $" + String.format("%.2f", totalSumOfAll));
+    }
+
+    public static void showCategory(int choice) {
+        Category c = chooseType(choice);
+        showSortInCategory(c.getType());
+    }
+
+    private static void showSortInCategory(String type) {
+        double totalSum = 0.00;
+        if (!noPurchasesAtAll()) {
+            for (Category c : ALL_PRODUCTS) {
+                if (c.getType().equals(type)) {
+                    System.out.println("\n" + c.getType() + ":");
+                    for (Product p : c.getPRODUCTS()) {
+                        totalSum += p.getPRICE();
+                    }
+                    c.getPRODUCTS().stream()
+                            .sorted(Comparator.comparing(Product::getPRICE).reversed())
+                            .forEach(System.out::println);
+                    System.out.println("Total sum: $" + String.format("%.2f", totalSum));
+                }
+            }
+        }
+    }
+
+    public static boolean noPurchasesAtAll() {
```

Comment:
Boolean method name usually starts with `is` - `isEmpty()`

Reply:
Done